test/integrationtests/voight_kampff/tools.py

```diff
@@ -23,8 +24,129 @@
 TIMEOUT = 10
 
 
+class CriteriaWaiter:
+    """Wait for a message to meet a certain criteria.
+
+    Args:
+        msg_type: message type to watch
+        criteria_func: Function to determine if a message fulfilling the
+                    test case has been found.
+        context: behave context
+    """
+    def __init__(self, msg_type, criteria_func, context):
+        self.msg_type = msg_type
+        self.criteria_func = criteria_func
+        self.context = context
+        self.result = Event()
+
+    def reset(self):
+        """Reset the wait state."""
+        self.result.clear()
+
+    # TODO: Remove in 21.08
+    def wait_unspecific(self, timeout):
+        """
+        Wait for a specified time for criteria to be fulfilled by any message.
+
+        This use case is deprecated and only for backward compatibility
```

Comment:
Do you know where this is or was being used? Or just an edge case that might exist?